tests/optim/test_single_node_adascale.py

```diff
@@ -82,14 +82,14 @@ def test_grad_accum(test_case, cpu):
         in_data_0 = Tensor(in_data[0])
         if not cpu:
             in_data_0 = in_data_0.cuda()
-        out = model(in_data_0)
-        out.sum().backward()
+        loss = model(in_data_0).sum() / 2
```

Comment:
in this test, can we add expected grad to adascale_test_data on line 63? That would explicitly check that the grad in all three cases below are the same:
1. ddp
2. grad with loss scale
3. grad without loss scale but with a final grad avg step

This should ensure there is no bug I suspected above in the step() function.